src/UniversalViewer.ts

```diff
@@ -63,22 +70,22 @@ export class UniversalViewer extends BaseContentHandler<IUVData<any>> {
       contentType = ContentType.IIIF;
     } else if (data[ContentType.YOUTUBE]) {
       contentType = ContentType.YOUTUBE;
-    } else if (this._contentType) {
-      contentType = this._contentType;
+    } else if (this.contentType) {
+      contentType = this.contentType;
     } else {
       contentType = ContentType.UNKNOWN;
     }
 
-    const handlerChanged: boolean = this._contentType !== contentType;
+    const handlerChanged: boolean = this.contentType !== contentType;
 
     if (contentType === ContentType.UNKNOWN) {
       console.error("Unknown content type");
     } else if (handlerChanged) {
-      this._contentType = contentType; // set content type
-      this._assignedContentHandler?.dispose(); // dispose previous content handler
+      this.contentType = contentType; // set content type
```

Comment:
For full backward compatibility:

```suggestion
      this.contentType = this._contentType = contentType; // set content type
```